### PR TITLE
feat: details popover for my company

### LINKED ISSUE #9214: Details popover for subsection
Prepare generic DetailsCellComponent, which will be able to pass details information 
image
[IMPORTANT!] We have another ticket to apply that component to all features

### LINKED ISSUE #9215: [Wait for #9214] Apply details button for all subsections
Do after #9214

Add details button in configuration for related features:

budgets:
- [x] cost centers
cost centers:
- [x] assigned budgets, manage budgets
units:
- [x] child units (what about active here?)
- [x] users (what about roles list here?)
- [x] assigned approvers, manage approvers (what about unit here?)
users:
- [x] assigned approvers, manage approvers
- [x] assigned user groups, manage user groups
- [x] assigned purchase limits, manage purchase limits
user groups:
- [x] assigned users, manage users
- [x] assigned purchase limits, manage purchase

### PR DESCRIPTION
**Pull request by Matejk00**
Closes: #9214

### CODE REVIEW

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  &.top-right > .arrow,
+  &.bottom-right > .arrow {
+    left: auto;
+    right: 2em;
+  }
+
+  &.left > .arrow,
+  &.right > .arrow {
+    top: 50%;
+    margin-top: -$cx-popover-arrow-width / 2;
```
Comment (morganm58) [inline projects/storefrontstyles/scss/components/misc/popover/_popover.scss:155]: Any padding and margin-left/right should be changed to -inline-start and -inline-end. 
`(padding or margin)-inline-start` for left 
`(padding or margin)-inline-end` for right

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+          ($cx-popover-arrow-width / 2);
+        border-bottom-color: $cx-popover-arrow-color;
+      }
+    }
+  }
+
+  &.top > .arrow,
+  &.bottom > .arrow {
+    left: 50%;
+    margin-left: -$cx-popover-arrow-width / 2;
```
Comment (morganm58) [inline projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: This should be `margin-inline-start: -$cx-popover-arrow-width / 2;`

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+          ($cx-popover-arrow-width / 2) 0;
+        border-right-color: $cx-popover-arrow-color;
+      }
+    }
+  }
+
+  &.left,
+  &.left-top,
+  &.left-bottom {
+    margin-right: $cx-popover-arrow-height;
```
Comment (morganm58) [inline projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: This should be `margin-inline-end: $cx-popover-arrow-height;`

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        border-width: $cx-popover-arrow-height ($cx-popover-arrow-width / 2) 0;
+        border-top-color: $cx-popover-arrow-color;
+      }
+    }
+  }
+
+  &.right,
+  &.right-top,
+  &.right-bottom {
+    margin-left: $cx-popover-arrow-height;
```
Comment (morganm58) [inline projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: Any padding and margin-left/right should be changed to -inline-start and -inline-end. This prevents it from breaking in multi-directional situations like the page direction changing from left to right and right to left
(padding or margin)-inline-start for left
(padding or margin)-inline-end for right

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss [hunk 5]
```diff
@@ -0,0 +1,174 @@
+$cx-popover-font-size: 0.875rem !default;
```
Comment (morganm58) [inline projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: When setting up variables we should follow the css-architecture guidelines in the Spartacus documentation found [here](https://sap.github.io/spartacus-docs/css-architecture/)
Example:
:root {
  --cx-color-primary: red;
}

cx-link {
  color: var(--cx-color-primary);
}

**Review by morganm58 — CHANGES_REQUESTED** [review]
I left some comments in _popover.scss

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss (hunk 2 above)
Comment (Matejk00) [reply projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: Updated

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss (hunk 3 above)
Comment (Matejk00) [reply projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: Updated

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss (hunk 4 above)
Comment (Matejk00) [reply projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: Updated

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss (hunk 5 above)
Comment (Matejk00) [reply projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: Updated

File: projects/storefrontlib/src/shared/components/popover/index.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export * from './popover.model';
+export * from './popover.component';
+export * from './popover.module';
+export * from './popover.service';
+export * from './positioning.service';
```
Comment (dunqan) [inline projects/storefrontlib/src/shared/components/popover/index.ts]: Shouldn't we also export the directive?

File: projects/storefrontlib/src/shared/components/popover/popover.directive.ts [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+  /**
+   * Method uses `Renderer2` service to listen every click event.
+   *
+   * Registered only when popover state was set to open and checks if element
+   * outside popover component was clicked.
+   *
+   * If so directive performs `close()` action and fire unlistener for such event.
+   */
+  triggerClickEvent() {
+    this.clickEventUnlistener = this.renderer.listen(
```
Comment (dunqan) [inline projects/storefrontlib/src/shared/components/popover/popover.directive.ts]: It's more like register (or listen) than trigger IMO.

File: projects/storefrontlib/src/shared/components/popover/popover.directive.ts [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+  constructor(
+    protected element: ElementRef,
+    protected viewContainer: ViewContainerRef,
+    protected componentFactoryResolver: ComponentFactoryResolver,
+    protected renderer: Renderer2,
+    protected changeDetectorRef: ChangeDetectorRef,
+    protected positioningService: PositioningService,
+    protected popoverService: PopoverService,
+    protected winRef: WindowRef,
+    @Inject(DOCUMENT) protected document: any
```
Comment (dunqan) [inline projects/storefrontlib/src/shared/components/popover/popover.directive.ts]: winRef already has Document token injected

File: projects/storefrontlib/src/shared/components/popover/popover.directive.ts [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+  ComponentRef,
+  Renderer2,
+  ChangeDetectorRef,
+  Output,
+  EventEmitter,
+  Inject,
+  HostListener,
+} from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+import { FocusConfig } from '../../../layout';
```
Comment (dunqan) [inline projects/storefrontlib/src/shared/components/popover/popover.directive.ts]: Not specific import

File: projects/storefrontlib/src/shared/components/popover/popover.directive.ts (hunk 8 above)
Comment (Matejk00) [reply projects/storefrontlib/src/shared/components/popover/popover.directive.ts]: ok, removed that from constructor

File: projects/storefrontlib/src/shared/components/popover/popover.directive.ts (hunk 7 above)
Comment (Matejk00) [reply projects/storefrontlib/src/shared/components/popover/popover.directive.ts]: no using this anymore

File: projects/storefrontlib/src/shared/components/popover/popover.directive.ts (hunk 9 above)
Comment (Matejk00) [reply projects/storefrontlib/src/shared/components/popover/popover.directive.ts]: updated

File: projects/storefrontlib/src/shared/components/popover/index.ts (hunk 6 above)
Comment (Matejk00) [reply projects/storefrontlib/src/shared/components/popover/index.ts]: good catch, added directive as well

**Review by developpeurweb — DISMISSED** [review]
Approving Keyboard Accessibility for Popover.

File: projects/storefrontstyles/scss/components/misc/popover/_popover.scss [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+  z-index: var(--cx-popover-z-index);
+  display: block;
+  max-width: var(--cx-popover-max-width);
+  font-size: var(--cx-popover-font-size);
+  word-wrap: break-word;
+  background-color: var(--cx-popover-background);
+  background-clip: padding-box;
+  border: var(--cx-popover-border-width) solid var(--cx-popover-border-color);
+  border-radius: var(--cx-popover-border-radius);
+  padding: 0.5rem;
```
Comment (morganm58) [inline projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: padding should be: `padding:1rem 0.938rem;`

Comment (Matejk00) [reply projects/storefrontstyles/scss/components/misc/popover/_popover.scss]: Updated

File: feature-libs/organization/administration/styles/_index.scss [hunk 11]
```diff
@@ -32,3 +32,20 @@ html[dir='rtl'] cx-org-list {
   --list-bg: url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='utf-8'%3F%3E%3Csvg xmlns='http://www.w3.org/2000/svg' version='1.1' viewBox='0 0 44.21 69.76' %3E%3Cpolyline transform='scale (-1, 1)' transform-origin='center' fill='none' stroke='%23D3D6DA' stroke-width='10' points='5,5 35,35 5,65 '/%3E%3C/svg%3E");
   --list-bg-pos: 25px;
 }
+
+.my-company-popover {
+  min-width: 200px;
```
Comment (morganm58) [inline feature-libs/organization/administration/styles/_index.scss]: After talking with Manu he said the min-width should be 140px and the popover should grow with the content. So we can remove the max-width and change the min-width to 140px

**Review by morganm58 — CHANGES_REQUESTED** [review]

File: feature-libs/organization/administration/styles/_index.scss (hunk 11 above)
Comment (Matejk00) [reply feature-libs/organization/administration/styles/_index.scss]: Updated.

**Review by morganm58 — DISMISSED** [review]

File: projects/storefrontlib/src/shared/components/popover/popover.directive.ts [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+      if (this.appendToBody) {
+        this.renderer.appendChild(
+          this.winRef.document.body,
+          this.popoverContainer.location.nativeElement
+        );
+      }
+
+      this.popoverContainer.changeDetectorRef.detectChanges();
+
+      setTimeout(() => this.handlePopoverEvents());
```
Comment (dunqan) [inline projects/storefrontlib/src/shared/components/popover/popover.directive.ts]: So you need two setTimeouts here? One for subscribe, one inside?

File: projects/storefrontlib/src/shared/components/popover/popover.directive.ts [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+   */
+  renderPopover() {
+    const containerFactory = this.componentFactoryResolver.resolveComponentFactory(
+      PopoverComponent
+    );
+    this.popoverContainer = this.viewContainer.createComponent(
+      containerFactory
+    );
+    if (this.popoverContainer && this.popoverContainer.instance) {
+      this.popoverContainer.instance.content = this.cxPopover;
```
Comment (dunqan) [inline projects/storefrontlib/src/shared/components/popover/popover.directive.ts]: Nitpick, what about 
`const componentInstance = this.popoverContainer.instance` 
and using this, to make it less boilerplated?

**Review by dunqan — DISMISSED** [review]

**Review by Pucek9 — APPROVED** [review]

**Pull request merged**